src/js/pages/chat.js

```diff
@@ -325,6 +327,85 @@ module.exports = BasePage.extend({
         }
         return false;
     },
+    handleMediaClick: function(e){
+        e.preventDefault();
+        var self = this;
+        console.log(this.model.topResource);
+        var file;
+        if (e.dataTransfer) {
+            file = e.dataTransfer.files[0];
+        } else if (e.target.files) {
+            file = e.target.files[0];
+        } else {
+            return;
+        }
+
+        if (file.type.match('image.*')) {
+            var downloadURL = URL.createObjectURL(file);
+            console.log('Offering:', file.name, file.size, file.type, file.lastModifiedDate,  downloadURL);
+            
+            var fileTracker = new FileReader();
+          
+            fileTracker.onload = function () {
+            console.log('file', file.name, file.size, file.type, file.lastModifiedDate);
+            var data =  this.result; 
+       /*     var resampler = new Resample(data, 80, 80, function (data) {*/
+            var b64Data = data.split(',')[1];
+            var id = crypto.createHash('sha1').update(atob(b64Data)).digest('hex');   
+            client.use(shareMedia);
+            var buf = Buffer.from(b64Data,'base64');
+                   
+            var message = {
+                    from:me.jid,
+                    id:client.nextId(),
+                    to: 'upload.hostname',  // replace hostname with your ejabberd host name         
```

Comment:
Could you please take it from the client configuration file?

[Here's an example](https://github.com/ForNeVeR/Kaiwa/blob/21d110646317e0563380818f5f31f7edf08f41cb/src/js/helpers/xmppEventHandlers.js#L77) (the parameters are stored in `dev_config.json`).

Reply:
@ForNeVeR , let me try

Reply:
i have added it